### PR TITLE
feat: rework placements of nav, and initialization of docsearch

### PR DESCRIPTION
**Pull request by harrysolovay**
* no more re-initializing docsearch on inputs (was creating excessive shadows, and using more & more memory -> performance breakdown)
* universal nav's search is perfectly centered (not just centered between its siblings)

Screen Shot 2020-06-05 at 11 06 51 AM
Screen Shot 2020-06-05 at 11 06 59 AM
Screen Shot 2020-06-05 at 11 07 04 AM
Screen Shot 2020-06-05 at 11 07 14 AM

### CODE REVIEW

**Comment by jordanranz** [comment]
Related to: https://github.com/aws-amplify/docs/pull/1915

File: client/src/docs-ui/search-bar/search-bar.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+@Component({tag: "docs-search-bar", shadow: false})
+export class DocsSearchBar {
+  initDocSearch() {
+    if (Build.isBrowser) {
+      // @ts-ignore
+      docsearch({
+        apiKey: ALGOLIA_API_KEY,
+        indexName: ALGOLIA_INDEX_NAME,
+        inputSelector: ".amplify-docs-search-input:not(.ds-input)",
```
Comment (jordanranz) [inline client/src/docs-ui/search-bar/search-bar.tsx:14]: Curious why this changed from `"#amplify-docs-search-input"`

File: client/src/docs-ui/search-bar/search-bar.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    this.initDocSearch();
+  }
+
+  render() {
+    return (
+      <Host class={searchStyle}>
+        <div>
+          <div>
+            <input
+              class="amplify-docs-search-input three-dee-effect"
```
Comment (jordanranz) [inline client/src/docs-ui/search-bar/search-bar.tsx:32]: Why did `amplify-docs-search-input` go from an id to a class?

File: client/src/docs-ui/universal-nav/universal-nav.tsx [hunk 3]
```diff
@@ -44,6 +45,10 @@ export class DocsUniversalNav {
               <sup>NEW</sup>
             </stencil-route-link>
 
+            <div class={searchContainerStyle}>
```
Comment (jordanranz) [inline client/src/docs-ui/universal-nav/universal-nav.tsx:48]: I don't think a container is needed here.

File: client/src/docs-ui/search-bar/search-bar.tsx (hunk 1 above)
Comment (harrysolovay) [reply client/src/docs-ui/search-bar/search-bar.tsx]: So that we never re-initialize an input as an Algolia docsearch element

**Comment by sammartinez** [comment]
testing locally, works as expected. working on reviewing the code now

File: client/src/docs-ui/search-bar/search-bar.style.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+// we use !important to override webkit's imposed border-radius
+export const searchStyle = css`
+  width: 100%;
+  ${MQLaptop} {
+    width: initial;
+  }
+  .algolia-autocomplete .ds-dropdown-menu {
+    min-width: initial;
+    ${MQLaptop} {
+      min-width: 500px;
```
Comment (sammartinez) [inline client/src/docs-ui/search-bar/search-bar.style.ts:13]: Shouldn't we have this be in rem?

File: client/src/docs-ui/search-bar/search-bar.tsx (hunk 2 above)
Comment (sammartinez) [reply client/src/docs-ui/search-bar/search-bar.tsx:37]: yes what is this reasoning?

Comment (harrysolovay) [reply client/src/docs-ui/search-bar/search-bar.tsx:37]: This was when we had two search bars on each page. I can no revert to an ID (thanks to Jordan's changes)

**Pull request merged**